bookkeeper-server/src/main/native/src/org/apache/bookkeeper/util/NativeIO.c

```diff
@@ -22,9 +22,17 @@
 #include <fcntl.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
-#include <asm-x86_64/unistd.h>
+#if !defined(_WIN32) && (defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__)))
+  /* UNIX-style OS. ------------------------------------------- */
+  #include <unistd.h>
+  #if (defined(__unit__) || defined(__unix)) && defined(__x86_64__)
+    #include <asm-x86_64/unistd.h>
+  #endif
+#endif
```

Comment:
This looks strange. If you include asm-x86_64/unistd.h do you still need unistd.h? 
Also based on the system unistd.h should point to the right arch based unistd.h isn't it?